src/neurostatslib/basis.py

```diff
@@ -652,16 +553,21 @@ def _evaluate(self, *sample_pts: NDArray) -> NDArray:
 
         Returns
         -------
-        basis_funcs : (number of basis, number of samples)
-            Raised cosine basis functions
+        basis_funcs :
+            Raised cosine basis functions, shape (number of basis, number of samples).
 
+        Raises
+        ------
+        ValueError
+            If the sample provided do not lie in [0,1].
         """
-
-        if any(sample_pts[0] < -1e-12) or any(sample_pts[0] > 1 + 1e-12):
+        if any(sample_pts < -np.finfo(sample_pts.dtype).resolution) or any(
+            sample_pts > 1 + np.finfo(sample_pts.dtype).resolution
```

Comment:
returning to this: where had you run into an issue with this? feels weird to check like this instead of just against 0/1, so I want to better understand where this came up

Reply:
still curious